tests/integration/supplychain/workload_reconciler_test.go

```diff
@@ -683,23 +687,50 @@ var _ = Describe("WorkloadReconciler", func() {
 								return []metav1.Condition{}
 							}
 
-							return workload.Status.Resources[1].Conditions
-						}).Should(MatchElements(getConditionOfType, IgnoreExtras, Elements{
+							return workload.Status.Resources[0].Conditions
+						}).Should(MatchAllElements(getConditionOfType, Elements{
 							"ResourceSubmitted": MatchFields(IgnoreExtras, Fields{
 								"Status":  Equal(metav1.ConditionFalse),
-								"Reason":  Equal("TemplateStampFailure"),
-								"Message": ContainSubstring("failed to recursively evaluate template: failed to interpolate template at path [data.foo]: evaluate tag $(config)$: jsonpath returned empty list: config"),
+								"Reason":  Equal(v1alpha1.NoHealthyStampedObjectsFoundReason),
+								"Message": ContainSubstring("unable to retrieve outputs for resource [my-first-resource] in supply chain [my-supply-chain]: failed to find any healthy object in the set of stamped objects"),
+							}),
+							"Healthy": MatchFields(IgnoreExtras, Fields{
+								"Status":  Equal(metav1.ConditionUnknown),
+								"Reason":  Equal("ReadyCondition"),
+								"Message": ContainSubstring("condition with type [Ready] not found on resource status"),
+							}),
+							"Ready": MatchFields(IgnoreExtras, Fields{
+								"Status": Equal(metav1.ConditionFalse),
+								"Reason": Equal(v1alpha1.NoHealthyStampedObjectsFoundReason),
 							}),
 						}))
 
 						workload := &v1alpha1.Workload{}
 						err := c.Get(ctx, client.ObjectKey{Name: "workload-joe", Namespace: testNS}, workload)
 						Expect(err).NotTo(HaveOccurred())
+
+						Expect(workload.Status.Conditions).To(MatchAllElements(getConditionOfType, Elements{
+							"SupplyChainReady": MatchFields(IgnoreExtras, Fields{
+								"Status": Equal(metav1.ConditionTrue),
+							}),
+							"ResourcesSubmitted": MatchFields(IgnoreExtras, Fields{
```

Comment:
I'm concerned by the confusion this might create in the top level conditions:
```
  Type: ResourcesSubmitted
  Status: False
  Reason: NoHealthyStampedObjectsFound
```

It reads to me like "nothing across the supply chain is healthy" (I can logically deduce that this is not what it means, but I'd still be forced to think about it)

Perhaps: 
```
  Type: ResourcesSubmitted
  Status: False
  Reason: NoHealthyStampedImmutableObjectFound
```